src/Cli/Microsoft.DotNet.Configurer/DotnetFirstTimeUseConfigurer.cs

```diff
@@ -39,17 +42,21 @@ public class DotnetFirstTimeUseConfigurer
             _reporter = reporter;
             _cliFallbackFolderPath = cliFallbackFolderPath;
             _pathAdder = pathAdder ?? throw new ArgumentNullException(nameof(pathAdder));
+            _performanceMeasurements ??= performanceMeasurements;
         }
 
         public void Configure()
         {
             if (ShouldAddPackageExecutablePath())
             {
+                DateTime beforeAddPackageExecutablePath = DateTime.Now;
                 AddPackageExecutablePath();
+                _performanceMeasurements?.Add("AddPackageExecutablePath Time", (DateTime.Now - beforeAddPackageExecutablePath).TotalMilliseconds);
```

Comment:
Instead of `DateTime.Now`, can you use a `Stopwatch` for all of the new measurements?  `Stopwatch` uses `QueryPerformanceCounter` which provides a much more fine grained measurement.  The cost should be ~the same as `Stopwatch` just captures a time stamp in the same way that this code does.